torch/csrc/autograd/python_variable.cpp

```diff
@@ -634,27 +638,59 @@ static PyObject* THPVariable_make_wrapper_subclass(PyObject*, PyObject* args, Py
   // data
   // TODO: for_blob produces non-resizable tensors, we might want this to be
   // resizable (have to define a custom allocator in that case)
-  auto data = at::for_blob(nullptr, r.intlist(1))
-        .strides(r.intlistOptional(2))
-        .storage_offset(r.toInt64Optional(3))
-        .context(nullptr, [](void *ctx) {})
-        .target_device(options.device())  // TODO: this shouldn't be necessary if it came from options
-        .options(options)
-        .make_tensor();
-  data.set_requires_grad(r.toBool(9));
-
-  const auto sizes_strides_policy = r.stringViewOptional(10);
-  if (sizes_strides_policy.has_value()) {
-    data.unsafeGetTensorImpl()->set_sizes_strides_policy(
-        parseSizesStridesPolicyArgument(*sizes_strides_policy));
+
+  Tensor tensor;
+  if (r.idx == 0) {
+    tensor = at::for_blob(nullptr, r.intlist(1))
+          .strides(r.intlistOptional(2))
+          .storage_offset(r.toInt64Optional(3))
+          .context(nullptr, [](void *ctx) {})
+          .target_device(options.device())  // TODO: this shouldn't be necessary if it came from options
+          .options(options)
+          .make_tensor();
+
+    const auto sizes_strides_policy = r.stringViewOptional(10);
+    if (sizes_strides_policy.has_value()) {
+      tensor.unsafeGetTensorImpl()->set_sizes_strides_policy(
+          parseSizesStridesPolicyArgument(*sizes_strides_policy));
+    }
+  } else {
+    AutoDispatchBelowADInplaceOrView guard{}; // TODO: Remove.
+    tracer::impl::NoTracerDispatchMode tracer_guard{};
+
+    // We shouldn't need storage
+    Storage storage{Storage::use_byte_size_t{}, 0, at::DataPtr{}};
+
+    tensor = at::detail::make_tensor<TensorImpl>(
+        std::move(storage), options.computeDispatchKey(), options.dtype());
+
+    auto sym_sizes = r.symintlist(1);
+    auto sym_strides = r.symintlist(2);
+
+    TensorImpl* tensor_impl = tensor.unsafeGetTensorImpl();
+
+    // TODO: this should probably be sym_sizes, sym_strides AND offset
+    tensor_impl->set_sym_sizes_and_strides(sym_sizes, sym_strides);
+
+    // TODO: this may need to be symbolic as well
+    auto storage_offset = r.toInt64Optional(3);
+    if (storage_offset) {
+      tensor_impl->set_storage_offset(*storage_offset);
+    }
+
+    // TODO: r.toBool(10)/dispatch_strides is ignored since setting sym_sizes/sym_strides
+    // will set a different custom policy
```

Comment:
Simple stopgap is to raise an error if these are set (in fact you could just remove it from the overload)